### PR TITLE
Modified unit test cases for SKABaseDevice and SKAMaster

### PR DESCRIPTION
**Pull request by adityadangeska**

### CODE REVIEW

File: skabase/SKABaseDevice/SKABaseDevice/SKABaseDevice.py [hunk 1]
```diff
@@ -442,6 +442,7 @@ def ToJson(self, argin):
         with exception_manager(self):
             defaults = {'with_value': False, 'with_commands': True,
                         'with_metrics': True, 'with_attributes': False}
+            #sys.stdout.write("Argin++++++" )
```
[the remark this answers is not in the document]
Comment (adityadangeska) [reply skabase/SKABaseDevice/SKABaseDevice/SKABaseDevice.py]: Yes. It was only for debugging. Will remove.

File: skabase/SKAMaster/test/SKAMaster_test.py [hunk 2]
```diff
@@ -94,15 +95,15 @@ def test_ToJson(self, tango_context):
...
+        assert tango_context.device.GetVersionInfo() == ["SKAMaster, tangods-skabasedevice, 1.0.0, A generic base device for SKA."]
         # PROTECTED REGION END #    //  SKAMaster.test_GetVersionInfo
 
     # PROTECTED REGION ID(SKAMaster.test_isCapabilityAchievable_decorators) ENABLED START #
     # PROTECTED REGION END #    //  SKAMaster.test_isCapabilityAchievable_decorators
     def test_isCapabilityAchievable(self, tango_context):
         """Test for isCapabilityAchievable"""
         # PROTECTED REGION ID(SKAMaster.test_isCapabilityAchievable) ENABLED START #
-        assert tango_context.device.isCapabilityAchievable([[0], [""]]) == False
+        assert tango_context.device.isCapabilityAchievable([[2], ["Cap1"]]) == False
```
[the remark this answers is not in the document]
Comment (adityadangeska) [reply skabase/SKAMaster/test/SKAMaster_test.py]: Not necessary to make it true. We can mimic like the mentioned capability is not achievable by returning false.

File: skabase/SKAMaster/test/SKAMaster_test.py [hunk 3]
```diff
@@ -151,15 +152,15 @@ def test_elementDatabaseAddress(self, tango_context):
     def test_buildState(self, tango_context):
         """Test for buildState"""
         # PROTECTED REGION ID(SKAMaster.test_buildState) ENABLED START #
-        assert tango_context.device.buildState == ''
+        assert tango_context.device.buildState == 'tangods-skabasedevice, 1.0.0, A generic base device for SKA.'
```
[the remark this answers is not in the document]
Comment (adityadangeska) [reply skabase/SKAMaster/test/SKAMaster_test.py]: This is theexact string which is being returned after unit test case is executed. Ideally, yes, it should be skamaster. Will check and modify accordingly.

File: skabase/SKABaseDevice/test/SKABaseDevice_test.py [hunk 4]
```diff
@@ -44,7 +45,7 @@ class TestSKABaseDevice(object):
     def mocking(cls):
         """Mock external libraries."""
         # Example : Mock numpy
-        # cls.numpy = SKABaseDevice.numpy = MagicMock()
+        #cls.numpy = SKABaseDevice.numpy = MagicMock()
```
[the remark this answers is not in the document]
Comment (adityadangeska) [reply skabase/SKABaseDevice/test/SKABaseDevice_test.py]: ok

File: skabase/SKAMaster/SKAMaster/SKAMaster.py [hunk 5]
```diff
@@ -118,7 +118,7 @@ def init_device(self):
         self._element_tel_state_device = ""
         self._element_database_device = ""
 
-        self._max_capabilities = {}
+        self._max_capabilities = {"cap1":10, "cap2":20}
```
[the remark this answers is not in the document]
Comment (adityadangeska) [reply skabase/SKAMaster/SKAMaster/SKAMaster.py]: yes. It is for testing. agree with your suggestion.

File: skabase/SKAMaster/SKAMaster/SKAMaster.py [hunk 6]
```diff
@@ -167,6 +167,7 @@ def read_maxCapabilities(self):
 
     def read_availableCapabilities(self):
         # PROTECTED REGION ID(SKAMaster.availableCapabilities_read) ENABLED START #
+        print (self._available_capabilities)
```
[the remark this answers is not in the document]
Comment (adityadangeska) [reply skabase/SKAMaster/SKAMaster/SKAMaster.py]: Yes. That was only for debugging.

File: skabase/SKAMaster/test/SKAMaster_test.py [hunk 7]
```diff
@@ -231,15 +232,15 @@ def test_testMode(self, tango_context):
...
+        assert tango_context.device.maxCapabilities == ('cap1:10', 'cap2:20') #('',)
         # PROTECTED REGION END #    //  SKAMaster.test_maxCapabilities
 
     # PROTECTED REGION ID(SKAMaster.test_availableCapabilities_decorators) ENABLED START #
     # PROTECTED REGION END #    //  SKAMaster.test_availableCapabilities_decorators
     def test_availableCapabilities(self, tango_context):
         """Test for availableCapabilities"""
         # PROTECTED REGION ID(SKAMaster.test_availableCapabilities) ENABLED START #
-        assert tango_context.device.availableCapabilities == ('',)
+        assert tango_context.device.availableCapabilities == ('cap1:10', 'cap2:20') #('',)
```
[the remark this answers is not in the document]
Comment (adityadangeska) [reply skabase/SKAMaster/test/SKAMaster_test.py]: done

File: skabase/SKABaseDevice/test/SKABaseDevice_test.py [hunk 8]
```diff
@@ -19,6 +19,7 @@
 from mock import MagicMock
 
 from PyTango import DevState
+import json
```
[the remark this answers is not in the document]
Comment (adityadangeska) [reply skabase/SKABaseDevice/test/SKABaseDevice_test.py]: done.

File: skabase/SKAMaster/test/SKAMaster_test.py [hunk 9]
```diff
@@ -231,15 +232,15 @@ def test_testMode(self, tango_context):
     def test_maxCapabilities(self, tango_context):
         """Test for maxCapabilities"""
         # PROTECTED REGION ID(SKAMaster.test_maxCapabilities) ENABLED START #
-        assert tango_context.device.maxCapabilities == ('',)
+        assert tango_context.device.maxCapabilities == ('cap1:10', 'cap2:20') #('',)
```
[the remark this answers is not in the document]
Comment (adityadangeska) [reply skabase/SKAMaster/test/SKAMaster_test.py]: done.

**Comment by ajoubertza** [comment]
@aditya  There are a number of comments that you have marked resolved/done, but I don't see any new commits on this PR.  Did you push your changes yet?

**Comment by adityadangeska** [comment]
> @aditya There are a number of comments that you have marked resolved/done, but I don't see any new commits on this PR. Did you push your changes yet?

Hi Anton,
I havent pushed my changes yet. I am busy with some other work.

**Pull request closed without merging**